File: m4dModels/SongMergeModel.cs
namespace m4dModels
{
    public class SongMergeModel(IEnumerable<SongHistory> songs)
    {
        public Guid SongId { get; set; } = Guid.NewGuid();
        public List<SongHistory> Songs { get; set; } = [.. songs];
    }
}

File: m4d/ClientApp/src/models/SongMergeModel.ts
import { jsonArrayMember, jsonMember, jsonObject } from "typedjson";
import { SongHistory } from "./SongHistory";

@jsonObject
export class SongMergeModel {
  @jsonMember public songId!: string;
  @jsonArrayMember(SongHistory) public songs!: SongHistory[];

  public constructor(init?: Partial<SongMergeModel>) {
    Object.assign(this, init);
  }
}
